### ISSUE TITLE
MAINT: consolidate data conversion interface/API

### ISSUE THREAD

**Issue by tylerjereddy** [darshan-hpc/darshan#490]
@jakobluettgau mentioned that we want to remove the dtype specification at `DarshanReport` creation time, favoring i.e., `to_df()` methods later on.

**Comment by shanedsnyder** [comment]
I'm a little confused on the `dtype` support. I mostly thought we had dropped that stuff for a single internal representation, with helper functions like `to_df()` for converting. Sounds like the option is not only still hanging around, but is triggering some slower code paths. Any reason not to just drop support for `dtype`, stick to a single preferred internal data representation, and offer `to_df()` style helpers for converting?

**Comment by nawtrey** [comment]
> Any reason not to just drop support for `dtype`, stick to a single preferred internal data representation, and offer `to_df()` style helpers for converting?

No, not that I'm aware of. I think this is what we are going to work towards at the moment. I know of some instances where `dtype` is being used that I'm planning on updating, I just haven't gotten to them yet.

**Comment by tylerjereddy** [comment]
Depends on whether you already have consumers--bigger projects might have to deprecate for a few releases before "just deleting", but if you don't have consumers of that API probably doesn't matter too much. If you're using semantic versioning that's an API break that isn't in an experimental namespace, so formally you'd want to bump the version accordingly, unless you consider it an actual bug.

**Comment by tylerjereddy** [comment]
Ok, that's up to you--I'm just relaying what I've seen considered a best practice for API changes on the Python side for stuff already in a public release. People and indeed automatic dependency resolvers should be able to trust that `3.x.x` -> `3.x.y` is API stable, etc.

**Comment by nawtrey** [comment]
As discussed in today's meeting, removing `dtype` will likely have little impact on users since it has been recommended to use the `DarshanRecordCollection` methods. So we can probably go ahead and remove it. Since this would be a pretty significant change, this should probably be added to a Change Log so users can see when the feature was removed. 

A few places this is likely still used in the repo: 
 - example jupyter notebooks
 - plotting code in `darshan/experimental/plots`

**Comment by kevin-harms** [comment]
I agree that we can remove dtype and make pandas the default. We can still have other to_* routines if other users would like to use those data representations.
